### PR TITLE
refactor(protocol-designer): add type exports to class-based components

### PR DESCRIPTION
**Pull request by mcous**
## overview

More types! This PR fully types the remaining components in PD, which were either:

- Class-based components
- HoCs that weren't built with react-redux `connect`

## changelog

- refactor(protocol-designer): add type exports to class-based components

## review requests

- Ensure changes are types only

## risk assessment

Very low if changes are types only

### CODE REVIEW

File: protocol-designer/src/components/SelectionRect.js [hunk 1]
```diff
@@ -79,7 +79,9 @@ export class SelectionRect extends React.Component<Props, State> {
     )
   }
 
-  getRect(args: DragRect) {
+  getRect(
+    args: DragRect
+  ): {| x0: number, x1: number, y0: number, y1: number |} {
```
Comment (IanLondon) [inline protocol-designer/src/components/SelectionRect.js:84]: I think this is `GenericRect` from `protocol-designer/src/collision-types.js` (which we should be able to make exact)

**Review by IanLondon — APPROVED** [review]
looks good!

File: protocol-designer/src/collision-types.js [hunk 2]
```diff
@@ -1,34 +1,24 @@
...
-export type BoundingRect = {
+export type BoundingRect = {|
   x: number,
   y: number,
   width: number,
   height: number,
-}
-
-export type ClientRect = {
-  // from flow's dom.js
```
Comment (mcous) [inline protocol-designer/src/collision-types.js:25]: This type was unused because, like this comment mentioned, it is built into Flow

**Review by Kadee80 — APPROVED** [review]
🔎

**Pull request merged**